test/sanity/pylint/ignore.txt

```diff
@@ -1,3 +1,5 @@
+lib/ansible/module_utils/compat_platform.py multiple-imports
+lib/ansible/module_utils/compat_platform.py ansible-format-automatic-specification
```

Comment:
Why are ignore/skip entries being added?

Reply:
In particular the `ansible-format-automatic-specification` ignore should not be used, as that rule indicates a Python 2.6 compatibility issue.

Reply:
The file started as a copy of 'platform.py' from python3.6  so I'm trying to keep it as close to upstream as possible for now. It already has a few tweaks to get it py2/py3 compatible though.

Sounds like I need to update the code that triggers ansible-format-automatic-specification errors as well to get it to be py2.6 compatible.

Reply:
I also need to add some unit tests to exercise it, since it passed 2.6 unit tests. I'll take a look a the coverage and see what it needs. Most of the module is unused at the moment, so it's possible tests covered some of it but not the py26 incompat parts though I expect it is just not covered in tests.

Reply:
The area of the code that failed that check, only happens on windows, so it might not be super easy to cover.

```diff
 +    # getwindowsversion() reflect the compatibility mode Python is
 +    # running under, and so the service pack value is only going to be
 +    # valid if the versions match.
 +    if winver[:2] == (maj, min):
 +        try:
 +            csd = 'SP{}'.format(winver.service_pack_major)
```

Reply:
A unit test with some mocking should be able to cover that, but probably not "super easy".

Reply:
@sivel And it looks like that code is only hit if using the compat_platform.win32_ver() which no code should be doing. Any code that needs that should continue to use the regular python platform.py since only the 'dist' and 'linux_distribution' methods are deprecated.

I can just remove all of the unused code here. 

I was thinking it might be easier to keep in sync with upstream pythons platform.py if we kept compat_platform.py as close as possible to upstream.

But now I think I'll just cut it down to just the code needed for the dist related functions.
I don't think keeping it exactly like upstream will buy us much. It's really not that much code in there that we need, and the upstream platform.py doesn't change very much (and the we need will be removed upstream after 3.7). And the code has already needed some py2/py3 compat tweaks, plus the ubuntu specific changes that isnt upstream.

ie, I'm going to remove all of the windows related code that fails sanity and anything else thats not needed.